### PR TITLE
Use only one log file for stdout

### PR DESCRIPTION
**Pull request by jkotalik**
For https://github.com/aspnet/IISIntegration/issues/1263

### CODE REVIEW

File: src/AspNetCoreModuleV2/CommonLib/FileOutputManager.cpp [hunk 1]
```diff
@@ -56,27 +60,21 @@ FileOutputManager::Start()
...
-    // Get the module name and add it to the log file name
-    // as two log files will be created, one from the shim
-    // and one from the request handler.
-    WCHAR path[MAX_PATH];
-    RETURN_LAST_ERROR_IF(!GetModuleFileName(g_hModule, path, sizeof(path)));
-    std::filesystem::path fsPath(path);
 
     // TODO fix string as it is incorrect
-    GetSystemTime(&systemTime);
+    RETURN_LAST_ERROR_IF(!GetProcessTimes(GetCurrentProcess(), &processCreationTime, &dummy1, &dummy2, &dummy3));
```
Comment (pakrym) [inline src/AspNetCoreModuleV2/CommonLib/FileOutputManager.cpp]: Can you pass the same variable or null?

File: src/AspNetCoreModuleV2/CommonLib/FileOutputManager.cpp [hunk 2]
```diff
@@ -62,7 +60,12 @@ FileOutputManager::Start()
 
 
     // TODO fix string as it is incorrect
```
Comment (pakrym) [inline src/AspNetCoreModuleV2/CommonLib/FileOutputManager.cpp]: TODO?

Comment (jkotalik) [reply src/AspNetCoreModuleV2/CommonLib/FileOutputManager.cpp]: Removed in other PR: https://github.com/aspnet/IISIntegration/pull/1332/files#diff-f02943abd05fa834ec7f4786668012a6L66

File: test/Common.FunctionalTests/Inprocess/LoggingTests.cs [hunk 3]
```diff
@@ -18,48 +17,43 @@ namespace Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests
...
-            {
-                await Helpers.AssertStarts(deploymentResult, path);
+            await Helpers.AssertStarts(deploymentResult, path);
 
-                StopServer();
+            StopServer();
 
-                var fileInDirectory = Directory.GetFiles(pathToLogs).Single(fileName => fileName.Contains("inprocess"));
-                var contents = File.ReadAllText(fileInDirectory);
+            var fileInDirectory = Directory.GetFiles(_logFolderPath).Single();
```
Comment (pakrym) [inline test/Common.FunctionalTests/Inprocess/LoggingTests.cs]: Replace all `Directory.GetFiles(_logFolderPath).Single();` calls with a helper method that uses exact file name

File: test/Common.FunctionalTests/Inprocess/LoggingTests.cs [hunk 4]
```diff
@@ -77,6 +71,30 @@ public async Task InvalidFilePathForLogs_ServerStillRuns()
...
+            await Helpers.AssertStarts(deploymentResult, "CheckLogFile");
+
+            StopServer();
+
+            var files = Directory.GetFiles(_logFolderPath);
+            Assert.Single(files);
+
+            var starttime = deploymentResult.HostProcess.StartTime.ToUniversalTime();
+            Assert.Equal(Path.Combine(_logFolderPath, $"std_{starttime.Year}{starttime.Month.ToString().PadLeft(2, '0')}" +
+                $"{starttime.Day.ToString().PadLeft(2, '0')}{starttime.Hour.ToString().PadLeft(2, '0')}" +
```
Comment (pakrym) [inline test/Common.FunctionalTests/Inprocess/LoggingTests.cs]: https://docs.microsoft.com/en-us/dotnet/standard/base-types/how-to-pad-a-number-with-leading-zeros

Comment (jkotalik) [reply test/Common.FunctionalTests/Inprocess/LoggingTests.cs]: Yeah that's about 3 million times better than what I did.

File: test/Common.FunctionalTests/Inprocess/LoggingTests.cs [hunk 5]
```diff
@@ -77,6 +71,22 @@ public async Task InvalidFilePathForLogs_ServerStillRuns()
...
+
+            deploymentParameters.EnableLogging(_logFolderPath);
+
+            var deploymentResult = await DeployAsync(deploymentParameters);
+            var logFileName = GetExpectedLogName(deploymentResult);
+            await Helpers.AssertStarts(deploymentResult, "CheckLogFile");
+
+            StopServer();
+
+            Assert.Equal(logFileName, GetActualLogName());
```
Comment (pakrym) [inline test/Common.FunctionalTests/Inprocess/LoggingTests.cs]: Assert.Single

File: test/IIS.FunctionalTests/Inprocess/StdOutRedirectionTests.cs [hunk 6]
```diff
@@ -98,14 +98,10 @@ public async Task EnableCoreHostTraceLogging_TwoLogFilesCreated()
...
 
-            var filesInDirectory = Directory.GetFiles(_logFolderPath);
-            Assert.Equal(2, filesInDirectory.Length);
-            foreach (var file in filesInDirectory)
-            {
-                var contents = File.ReadAllText(file);
-                EventLogHelpers.VerifyEventLogEvent(deploymentResult, TestSink, "Invoked hostfxr");
-                Assert.Contains("Invoked hostfxr", contents);
-            }
+            var fileInDirectory = Directory.GetFiles(_logFolderPath).Single();
```
Comment (pakrym) [inline test/IIS.FunctionalTests/Inprocess/StdOutRedirectionTests.cs]: Use helper

File: test/Common.FunctionalTests/Inprocess/LoggingTests.cs [hunk 7]
```diff
@@ -18,48 +17,43 @@ namespace Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests
...
-            deploymentParameters.WebConfigActionList.Add(
-                WebConfigHelpers.AddOrModifyAspNetCoreSection("stdoutLogEnabled", "true"));
-
-            var pathToLogs = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
-            deploymentParameters.WebConfigActionList.Add(
-                WebConfigHelpers.AddOrModifyAspNetCoreSection("stdoutLogFile", Path.Combine(pathToLogs, "std")));
+            deploymentParameters.EnableLogging(_logFolderPath);
 
             var deploymentResult = await DeployAsync(deploymentParameters);
+            var logFileName = GetExpectedLogName(deploymentResult);
```
Comment (pakrym) [inline test/Common.FunctionalTests/Inprocess/LoggingTests.cs]: Call process.StartTime in deployers to cache it https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.Process/src/System/Diagnostics/Process.cs#L192

**Review by pakrym — APPROVED** [review]

**Pull request merged**